cdap-cli/src/main/java/co/cask/cdap/cli/command/security/AddRoleToPrincipalCommand.java

```diff
@@ -47,13 +47,12 @@ public void perform(Arguments arguments, PrintStream output) throws Exception {
     String principalName = arguments.get("principal-name");
     client.addRoleToPrincipal(new Role(roleName), new Principal(principalName, Principal.PrincipalType.valueOf
       (principalType.toUpperCase())));
-    output.printf("Successfully added role '%s' to %s %s\n", roleName, principalType, principalName);
+    output.printf("Successfully added role '%s' to '%s' '%s'\n", roleName, principalType, principalName);
   }
 
   @Override
   public String getPattern() {
-    return String.format("add role <%s> to %s <%s> %s <%s>", ArgumentName.ROLE_NAME,
-                         ArgumentName.PRINCIPAL_TYPE, ArgumentName.PRINCIPAL_TYPE, ArgumentName.PRINCIPAL_NAME,
+    return String.format("add role <%s> to <%s> <%s>", ArgumentName.ROLE_NAME, ArgumentName.PRINCIPAL_TYPE,
```

Comment:
I didn't mean to add the check in the CLI. But it doesn't seem like something every extension should do. I think we should add it in the handler. I don't think we should allow adding a role to another unless we have a very specific use case for it.

Its ok if you don't want to take care of this in this PR, but lets report a bug to fix this.